crates/fluvio-sc-schema/src/shared.rs

```diff
@@ -0,0 +1,98 @@
+pub type Result = std::result::Result<(), ValidateResourceNameError>;
+
+pub const MAX_RESOURCE_NAME_LEN: usize = 63;
+
+#[derive(Copy, Clone, Debug)]
+pub enum ValidateResourceNameError {
+    NameLengthExceeded,
+    InvalidCharacterEncountered,
+}
+
+/// Checks if the Resource Name is valid for internal resources.
+///
+/// ```test
+/// let name = "prices-list-scrapper";
+/// assert!(is_valid_resource_name(name));
+/// ```
+pub fn is_valid_resource_name(name: &str) -> Result {
```

Comment:
since this return error, it should be something like: `check_resource_name`

Reply:
Fixed!